resources/js/openreplay.js

```diff
@@ -3,11 +3,73 @@ import Tracker from '@openreplay/tracker'
 const captureNetwork = import.meta.env.VITE_OPENREPLAY_CAPTURE_NETWORK === undefined
     || import.meta.env.VITE_OPENREPLAY_CAPTURE_NETWORK === 'true'
 
+// https://docs.openreplay.com/en/installation/network-options/
+const openreplaySanitizer = (requestAndResponse) => {
+    for (const headerToSanitize of ['Authorization', 'X-CSRF-Token']) {
+        if (requestAndResponse.request.headers[headerToSanitize]) {
+            requestAndResponse.request.headers[headerToSanitize] = '***'
+        }
+    }
+
+    if (typeof requestAndResponse.request.body !== 'string') {
+        // If the request body is not a string, we do not save it.
+        // As this will be one of https://developer.mozilla.org/en-US/docs/Web/API/RequestInit#body
+        requestAndResponse.request.body = ''
+    }
+
+    let responseBodyString = requestAndResponse.response.body
+    if (typeof responseBodyString === 'object') {
+        try {
+            responseBodyString = JSON.stringify(responseBodyString)
+        } catch (e) {
+            // Rather safe than sorry, if it fails we empty the response body
+            responseBodyString = '{}'
+        }
+    }
+
+    for (const jsonValuesToSanitize of [
+        'password', 
+        'token',
+        'mask',
+        'cart_id',
+        'postcode',
+        'city',
+        'customer_telephone',
+        'telephone',
+        'fax',
+        'vat_id',
+        'po_number',
+        'pay_return_url',
+        'pay_redirect_url',
+        'iban'
+    ]) {
```

Comment:
Should we make this list easy to extend? Maybe with a config file?

Reply:
We could but it'd be safer to keep the `window.openreplaySanitizer` for that and keeping everything for the first pass hardcoded.

Allowing extending via a config file means we'd have to pass that through via a public js variable, which any script on the site could change and use to escape the sanitization regex.

Vite does compile .env values directly to their value when building, so that should be safe enough to implement